web/src/avatar.js

```diff
@@ -34,11 +35,32 @@ export function build_bot_create_widget() {
     );
 }
 
-export function build_bot_edit_widget($target) {
+export function build_bot_edit_widget($target, bot_details) {
     const get_file_input = function () {
         return $target.find(".edit_bot_avatar_file_input");
     };
 
+    if (bot_details.avatar_source === "G") {
+        $("#edit-bot-avatar-upload-widget .image-hover-background").hide();
+        $("#edit-bot-avatar-upload-widget .image-delete-button").hide();
+    }
+
+    $("#edit-bot-avatar-upload-widget .image-delete-button").on("click", (e) => {
+        e.preventDefault();
+        e.stopPropagation();
+        $("#edit-bot-avatar-upload-widget .upload-spinner-background").css({visibility: "visible"});
+        $("#edit-bot-avatar-upload-widget .image-hover-background").hide();
+        $("#edit-bot-avatar-upload-widget .image-delete-button").hide();
```

Comment:
This duplication can be avoided by refactoring `display_avatar_delete_complete` to accept a container element as parameter and we can pass `$("#user-avatar-upload-widget")` and `$("#edit-bot-avatar-upload-widget")` to it. This should be done in a separate prep commit.

Reply:
I will create a follow-up pull request to display a loading spinner in the bot edit widget when the current avatar is deleted. Just refactoring the `display_avatar_delete_complete` function won't work, because for the profile picture, it was called on the success of the '/json/users/me/avatar' endpoint's call. However, in the bot avatar widget, when the delete button is clicked, only the source of the image element of `#current_bot_avatar_image` is changed to the gravatar URL. I will tackle this by making it asynchronous and calling `display_avatar_delete_complete` once the image source is updated.

or Should I do this in a separate commit within this pull request?

Reply:
I am not sure whether we would need a spinner here since the request to delete the avatar will be sent only after clicking "Save" button as you mentioned, so we can leave that for now. 

Might need some changes if we decide on some UI/UX changes further, but looks fine for now.